functions/Copy-SqlDatabaseMail.ps1

```diff
@@ -115,34 +115,60 @@ Shows what would happen if the command were executed.
     begin {
 
         function Copy-SqlDatabaseMailConfig {
-            Write-Message -Message "Migrating mail server configuration values" -Level Output -Silent $Silent
+            [cmdletbinding(SupportsShouldProcess = $true)]
+            param()
+
+            Write-Message -Message "Migrating mail server configuration values" -Level Verbose -Silent $Silent
+            $copyMailConfigStatus = [pscustomobject]@{
+                SourceServer = $sourceServer
+                DestinationServer = $destServer
+                Name = "Mail Configuration"
+                Type = "Configuration"
+                Status = $null
+                DateTime = [sqlcollective.dbatools.Utility.DbaDateTime](Get-Date)
+            }
             if ($pscmdlet.ShouldProcess($destination, "Migrating all mail server configuration values")) {
                 try {
                     $sql = $mail.ConfigurationValues.Script() | Out-String
                     $sql = $sql -replace [Regex]::Escape("'$source'"), [Regex]::Escape("'$destination'")
                     Write-Message -Message $sql -Level Debug
                     $destServer.ConnectionContext.ExecuteNonQuery($sql) | Out-Null
                     $mail.ConfigurationValues.Refresh()
+                    $copyMailConfigStatus.Status = "Successful"
                 }
                 catch {
+                    $copyMailConfigStatus.Status = "Failed"
+                    $copyMailConfigStatus
                     Stop-Function -Message "Unable to migrate mail configuration" -Category InvalidOperation -InnerErrorRecord $_ -Target $destServer -Silent $Silent
                 }
             }
+            $copyMailConfigStatus
         }
 
         function Copy-SqlDatabaseAccount {
             $sourceAccounts = $sourceServer.Mail.Accounts
             $destAccounts = $destServer.Mail.Accounts
 
-            Write-Message -Message "Migrating accounts" -Level Output -Silent $Silent
+            Write-Message -Message "Migrating accounts" -Level Verbose -Silent $Silent
```

Comment:
The only thing I'd change is this -Silent $Silent is not needed since it's inherited (thats why you add it at the top)